mailpoet/lib/Automation/Integrations/MailPoet/Analytics/Endpoints/OverviewEndpoint.php

```diff
@@ -0,0 +1,67 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Analytics\Endpoints;
+
+use MailPoet\API\REST\Request;
+use MailPoet\API\REST\Response;
+use MailPoet\Automation\Engine\API\Endpoint;
+use MailPoet\Automation\Engine\Exceptions\NotFoundException;
+use MailPoet\Automation\Engine\Exceptions\RuntimeException;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+use MailPoet\Automation\Integrations\MailPoet\Analytics\Controller\OverviewStatisticsController;
+use MailPoet\Automation\Integrations\MailPoet\Analytics\Entities\Query;
+use MailPoet\Validator\Builder;
+
+class OverviewEndpoint extends Endpoint {
+
+  /** @var AutomationStorage */
+  private $automationStorage;
+
+  /** @var OverviewStatisticsController */
+  private $overviewStatisticsController;
+
+  public function __construct(
+    AutomationStorage $automationStorage,
+    OverviewStatisticsController $overviewStatisticsController
+  ) {
+    $this->automationStorage = $automationStorage;
+    $this->overviewStatisticsController = $overviewStatisticsController;
+  }
+
+  public function handle(Request $request): Response {
+    $automationid = $request->getParam('id');
+    if (!is_int($automationid)) {
+      throw new RuntimeException('Invalid automation id');
+    }
```

Comment:
You can just cast this to `int`, the request schema is validated.

Reply:
62937c83c